### PR TITLE
Rishabh sharma code review

### PR DESCRIPTION
**Pull request by sharmrb**
**Analysis of the program.**
The application is useful for coordinating group meetings. This application takes out the hassle of asking everyone and deciding the time, Rather you can create blocks on the time you're busy in the in-app calendar in this application and it can be synced with other members of the team so everyone can meet at a time that works.

**Was the program available in UC Github on time?**
Yes The program was available to me right when I asked for access.

**Is the program documented/commented well enough for you to understand?**
The program was not as well documented, A lot of empty classes with short descriptions, which was a little confusing for me.

**Does the program compile?**
Yes, the program complied.

**Rationale behind your changes.**
My Changes were based on how I can help reduce the technical debt, So a few changes or additions I made was adding a da atabase class and creating database manager class, Which can help this team manage it's database better and have a better structure. I also created auto-assigned primary key ID to events , just as an example so that the team can replicate it and code it to have events added to the database. I also made some minor changes that I thought were causing an error such as adding an @requiresApi annotation which was causing an error in .now() functions.

**Links to three specific commits that you made to your own group's GitHub repository before the end of sprint deadline.**  
[bc22d7bb7591c55777df792d59cf2c11fd14915f]
https://github.com/Odogwufu/TimeMan2/commit/bc22d7bb7591c55777df792d59cf2c11fd14915f

[fe6c3bfd292fe75e0040e10c3243eadd75cb36e2]
https://github.com/Odogwufu/TimeMan2/commit/fe6c3bfd292fe75e0040e10c3243eadd75cb36e2

[5304375b55c041cff2a763f659437302f9525155]
https://github.com/Odogwufu/TimeMan2/commit/5304375b55c041cff2a763f659437302f9525155

I had 23 Commits made before the end of sprint 2, These 3 commits were what they led up to and all of them can be seen on my branch on my team's project.
 
Next, list three specific technical concepts that you learned from this code review.  Copy and paste samples to demonstrate your point. 
1. How to implement a working calendar, we plan to do the same with our project soon and it is a good example
2. The best practices used, Such as how to create a class header with long parameters, They did a great job of following them.
3. How to have a great UI that utilizes the whole screen, Their activity page is just the right amount of dense and has good amount of functions, I like that so much was included on a single screen and would help a user do most actions right from that page.

### CODE REVIEW

File: app/src/main/java/com/plan2meet/dto/EventDTO.kt [hunk 1]
```diff
@@ -14,4 +17,12 @@ class EventDTO (
     override fun toString(): String {
         return startTime.toString()
     }
+
+    @JvmField
+    @PrimaryKey(autoGenerate = true)
+    var id: Long = 0
+
+    fun UpdateDb(){
```
Comment (sharmrb) [inline app/src/main/java/com/plan2meet/dto/EventDTO.kt:25]: An Update Db function could be created here which can use the database manager class

File: app/src/main/java/com/plan2meet/dto/EventDTO.kt [hunk 2]
```diff
@@ -14,4 +17,12 @@ class EventDTO (
     override fun toString(): String {
         return startTime.toString()
     }
+
+    @JvmField
+    @PrimaryKey(autoGenerate = true)
+    var id: Long = 0
```
Comment (sharmrb) [inline app/src/main/java/com/plan2meet/dto/EventDTO.kt:23]: I added this as an example of how data can be put into the table

File: app/src/main/java/com/plan2meet/dto/EventDTO.kt [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
 
 import android.os.Build
+import android.os.Build.VERSION_CODES
 import androidx.annotation.RequiresApi
 import java.time.LocalDateTime
 import java.time.LocalDateTime.now
 
-@RequiresApi(Build.VERSION_CODES.26)
+@RequiresApi(VERSION_CODES.26)
 class EventDTO (
```
Comment (sharmrb) [inline app/src/main/java/com/plan2meet/dto/EventDTO.kt:13]: I saw an error that said the .now() function require API level 26 , So i added the annotation. It fixed the error

File: app/src/main/java/com/plan2meet/ui/MainActivity.kt [hunk 4]
```diff
@@ -9,8 +9,8 @@ class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         val event = EventDTO();
-        print(event.toString())
-        setContentView(R.layout.activity_main)
+        print(event.toString());
+        setContentView(R.layout.activity_main);
 
     }
```
Comment (sharmrb) [inline app/src/main/java/com/plan2meet/ui/MainActivity.kt:15]: This can be ignored.

File: app/src/main/java/com/plan2meet/dao/DatabaseManager.kt [hunk 5]
```diff
@@ -0,0 +1,4 @@
+package com.plan2meet.dao
+
+class DatabaseManager {
```
Comment (sharmrb) [inline app/src/main/java/com/plan2meet/dao/DatabaseManager.kt:3]: I would recommend creating a database manager class to access DB anywhere in the project.

File: app/src/main/java/com/plan2meet/AppModule.kt [hunk 6]
```diff
@@ -1,2 +1,2 @@
-package com.example.plan2meet
+package com.plan2meet
```
Comment (discospiff) [inline app/src/main/java/com/plan2meet/AppModule.kt:1]: Good.  com.example is probably not a domain name owned by this group.  :)

File: app/src/main/java/com/plan2meet/dao/DatabaseManager.kt (hunk 5 above)
Comment (discospiff) [reply app/src/main/java/com/plan2meet/dao/DatabaseManager.kt:3]: This looks like an incomplete feature.
Are you using these anywhere?  I don't see them used.
If not in use, remove them.  Keep the codebase as small as possible to do the job you want to do.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.

File: app/src/main/java/com/plan2meet/dto/EventDTO.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/com/plan2meet/dto/EventDTO.kt:25]: This looks like an incomplete feature.
Are you using these anywhere?  I don't see them used.
If not in use, remove them.  Keep the codebase as small as possible to do the job you want to do.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.

File: app/src/main/java/com/plan2meet/ui/AppDatabase.kt [hunk 7]
```diff
@@ -0,0 +1,5 @@
+package com.plan2meet.ui
+
+class AppDatabase  {
+
+}
```
Comment (discospiff) [inline app/src/main/java/com/plan2meet/ui/AppDatabase.kt:5]: This looks like an incomplete feature.
Are you using these anywhere?  I don't see them used.
If not in use, remove them.  Keep the codebase as small as possible to do the job you want to do.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.

File: app/src/main/java/com/plan2meet/ui/GroupListProvider.kt [hunk 8]
```diff
@@ -0,0 +1,5 @@
+package com.plan2meet.ui
+
+class GroupListProvider {
+
+}
```
Comment (discospiff) [inline app/src/main/java/com/plan2meet/ui/GroupListProvider.kt:5]: This looks like an incomplete feature.
Are you using these anywhere?  I don't see them used.
If not in use, remove them.  Keep the codebase as small as possible to do the job you want to do.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.

**Comment by discospiff** [comment]
The suggestions in this review  are likely needed for the project at some point, but are new, and sometimes incomplete, functionality.  New functionality is better in a separate branch until fully developed, as features should not be merged to main/master until the meet the team's Definition of Done.  Code review branches should minimize technical debt, and thus, be merge-able to master immediately.  So, if the team desires these features, I'd keep the branch open, and iterate over the feature until it is complete.

**Comment by WillTaylor2352** [comment]
As of right now, we will be proceeding with development without using the changes proposed in this pull request

**Pull request closed without merging**